codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustWriter.kt

```diff
@@ -475,18 +474,44 @@ class RustWriter private constructor(
                 }
             }
 
-            shape is NumberShape -> rustBlock("if ${outerField.removePrefix("&")} != 0") {
+            else -> this.block(outerField)
+        }
+    }
+
+    /**
+     * Generate a wrapping if statement around a primitive field.
+     * The specified block will only be called if the field is not set to its default value - `0` for
+     * numbers, `false` for booleans.
+     */
+    fun ifNotDefault(shape: Shape, outerField: String, block: RustWriter.(field: String) -> Unit) {
```

Comment:
It seems to me like this or `software.amazon.smithy.rust.codegen.core.smithy.protocols.serialize.SerializerUtil` could be made DRYer now.